### PR TITLE
fix(HTTP Request Node): Ignore empty body for auto detect json

### PR DESCRIPTION
**Pull request by feelgood-interface**
Github issue / Community forum post (link here to close automatically):
https://github.com/n8n-io/n8n/issues/5214

### CODE REVIEW

File: packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts [hunk 1]
```diff
@@ -1347,7 +1347,8 @@ export class HttpRequestV3 implements INodeType {
 				const responseContentType = response.headers['content-type'] ?? '';
 				if (responseContentType.includes('application/json')) {
 					responseFormat = 'json';
-					response.body = jsonParse(Buffer.from(response.body).toString());
+					const data = Buffer.from(response.body).toString();
+					response.body = !data ? undefined : jsonParse(data);
```
Comment (netroy) [inline packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts:1351]: We should not be changing the default behavior here, as that would mean that if an API starts accidentally returning empty bodies, n8n would silently ignore the issue.

You could combine this check with the option `neverError`, to only ignore the error when a user has explicitly asked not to throw an error.

```suggestion
					const neverError = this.getNodeParameter(
						'options.response.response.neverError',
						0,
						false,
					) as boolean;
					const data = Buffer.from(response.body).toString();
					response.body = jsonParse(data, {
						...(neverError
							? { fallbackValue: {} }
							: { errorMessage: 'Invalid JSON in response body' }),
					});
```

**Review by Joffcom — APPROVED** [review]
Tested and works well.

**Comment by Joffcom** [comment]
Hey @feelgood-interface,

Thanks for the PR 🙌🏻 , We made a bit of a change so that there is still an error to let you know the JSON is invalid and if you want to ignore the error you can set Options > Response > Never Error and that will now correctly allow the workflow to continue.

Thanks for taking the time to put this one in.

**Comment by janober** [comment]
Got released with `n8n@0.216.0`

**Pull request merged**